### PR TITLE
Extract EXIF parser from OpenCV

### PR DESCRIPTION
**Pull request by havaker**
## Category:

**New feature** (*non-breaking change which adds functionality*)

## Description:

Many image formats may contain EXIF metadata. To support parsing this metadata I've added an EXIF parser from OpenCV to our `third_party/` directory.

### Affected modules and functionalities:

Added a new module to the `third_party` directory.

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: third_party/opencv/exif/exif.cc [hunk 1]
```diff
@@ -0,0 +1,553 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (JanuszL) [inline third_party/opencv/exif/exif.cc:1]: I'm not sure if we can add our license on top. I would at most extend the copyright of the original file.

Comment (JanuszL) [reply third_party/opencv/exif/exif.cc:1]: If it is more than just a copy paste then it should be fine to do it this way.

Comment (mzient) [reply third_party/opencv/exif/exif.cc:1]: The libtiff decoder has been substantially altered. Here, we're barely introducing some mechanical changes. I think the changes we've introduced don't warrant sublicensing.

Comment (havaker) [reply third_party/opencv/exif/exif.cc:1]: In that case, I will remove our license blurb.

**Comment by havaker** [comment]
!build

**Review by jantonguirao — APPROVED** [review]

**Review by mzient — APPROVED** [review]

**Pull request merged**